ocaml/typing/env.ml

```diff
@@ -2931,18 +2940,29 @@ let lock_mode ~errors ~loc env id vmode locks =
                 (Local_value_used_in_closure (id, escaping_context))
         end
       | Exclave_lock ->
-        match Value_mode.submode vmode Value_mode.regional with
-        | Ok () -> Value_mode.regional_to_local vmode
-        | Error _ ->
+        begin
+          match Value_mode.submode vmode Value_mode.regional with
+          | Ok () -> Value_mode.regional_to_local vmode
+          | Error _ ->
+            may_lookup_error errors loc env
+              (Local_value_used_in_exclave id);
+        end
+      | Unboxed_lock ->
```

Comment:
I think it would be slightly better if this function just returned a bool (or better a variant type) indicating whether the lookup had crossed an `Unboxed_lock`. This can then be returned as far as the `lookup_value` function -- where we already force the value description -- and the check of the layout performed there. (I would probably just ignore the lock in the `find_by_name` function). I think this would make it clearer that the forcing of the value description doesn't matter.